zipkin-storage/cassandra3/src/main/java/zipkin/storage/cassandra3/CassandraUtil.java

```diff
@@ -96,14 +97,17 @@ public static int durationIndexBucket(long ts_micro) {
       return Collections.emptyList();
     }
     checkArgument(request.serviceName != null, "serviceName needed with annotation query");
-    Set<String> annotationKeys = new LinkedHashSet<>();
```

Comment:
LinkedHashSet retains order.. how about using that?

On Mon, Sep 12, 2016 at 11:16 AM, mck <notifications@github.com> wrote:

> In zipkin-storage/cassandra3/src/main/java/zipkin/storage/
> cassandra3/CassandraUtil.java
> <https://github.com/openzipkin/zipkin/pull/1289#discussion_r78311519>:
>
> > @@ -96,14 +97,17 @@ public static int durationIndexBucket(long ts_micro) {
> >        return Collections.emptyList();
> >      }
> >      checkArgument(request.serviceName != null, "serviceName needed with annotation query");
> > -    Set<String> annotationKeys = new LinkedHashSet<>();
>
> went to list with manual dedupe to keep order. (i think the reason was to
> have some consistency when reading the data in cqlsh. ie you're always
> reading it as service, annotations, binaryAnnotations)
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/openzipkin/zipkin/pull/1289/files/ded65ab27c2839701ce7ac523bcf8f4df79a3526#r78311519>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAD6175ia4jL2tNbs_IWjREzwvwvkTLbks5qpMQKgaJpZM4J5_ci>
> .
>